Tests/ApodiniTests/ApodiniTypeDescriptorTests/TypeDescriptorTests.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  File.swift
+//  
+//
+//  Created by Eldi Cano on 12.04.21.
+//
+
+@testable import ApodiniTypeDescriptor
+import XCTest
+
+
+class TypeDescriptorTests: XCTestCase {
+    struct Car: Encodable {
+        let plateNumber: Int?
+        let color: String
+    }
+    
+    struct User: Encodable {
+        private enum CodingKeys: String, CodingKey {
+            case id, name = "_name", car
+        }
+        
+        let id: Int?
+        let name: String
+        let car: Car?
+    }
+    
+    
+    struct Student: Encodable {
+        let id: Int?
+        let users: [String:User]
+        let grades: [Float]
+        let name: String?
+        let surname: String
+    }
+    
+    func testStudentTypeDescriptor() throws {
+        let descriptor = try typeDescriptor(Student.self)
+        
+        XCTAssertEqual(descriptor.debugDescription, expectedResult().debugDescription)
```

Comment:
Looks about right. I would suggest to add a test with a custom-conformance to `Codable`, though.